lm_eval/models/huggingface.py

```diff
@@ -147,18 +149,24 @@ def __init__(
             assert isinstance(pretrained, str)
             assert isinstance(batch_size, (int, str))
 
-            gpus = torch.cuda.device_count()
+            device_counts = torch.cuda.device_count()
+
             accelerator_kwargs = InitProcessGroupKwargs(timeout=timedelta(weeks=52))
             accelerator = Accelerator(kwargs_handlers=[accelerator_kwargs])
+
             if accelerator.num_processes > 1:
                 self.accelerator = accelerator
 
+            if is_npu_available():
```

Comment:
instead of `is_npu_available()` what about doing `if "npu" in device` ?

Reply:
replace `if is_npu_available` with `if "npu" in accelerator.device.type`, 
because `accelerate` can automatically choose to use cuda or npu when they are available